frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/redux/validations/vm-settings-tab-validation.ts

```diff
@@ -95,6 +103,37 @@ export const validateOperatingSystem: VmSettingsValidator = (field) => {
   return asValidationObject(`Select matching for: ${guestFullName}`, ValidationErrorType.Info);
 };
 
+const memoryValidation: VmSettingsValidator = (field, options) => {
+  const memValueGB = iGetFieldValue(field);
+  if (!memValueGB) {
+    return null;
+  }
+  const memValueBytes = memValueGB * 1024 ** 3;
+  const validations = getTemplateValidations(options, getFieldId(VMSettingsField.MEMORY));
+  if (validations.length === 0) {
+    return null;
+  }
+
+  /*
+  - Check if at least one validation has passed.
+  - Would *NOT* check if all the validations have passed since it may not be possible
+    For example:
+    If we have two 'between' validations: 2-4 and 6-10 from 2 different templates,
+    There is no value that would satisfy both validations.
+  */
+  let errorMsg;
+  const isValid = validations.some((validation) => {
```

Comment:
can we do ?
```
 const ( isValid, message } = runValidations(validations, memValueBytes);
```
see https://github.com/openshift/console/pull/3839/files#r363104852

Reply:
Or we could create a Validation class which would encapsulate all of it - validation data (on initialization) and the logic